### PR TITLE
add `speedscope`

### PR DESCRIPTION
**Pull request by moritzwilksch**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by moritzwilksch** [comment]
@conda-forge/help-nodejs is this how you package npm packages?

File: recipes/speedscope/build.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+export npm_config_build_from_source=true
+
+rm $PREFIX/bin/node
+ln -s $BUILD_PREFIX/bin/node $PREFIX/bin/node
+
+NPM_CONFIG_USERCONFIG=/tmp/nonexistentrc
+npm install -g speedscope
+
+pnpm import
+pnpm install --prod
```
Comment (xhochy) [inline recipes/speedscope/build.sh:18]:
```suggestion
pnpm import
pnpm install --prod
pnpm pack
npm install -g speedscope-${PKG_VERSION}.tar.gz
```

File: recipes/speedscope/build.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+if [[ "${target_platform}" == "osx-arm64" ]]; then
+    export npm_config_arch="arm64"
+fi
+# Don't use pre-built gyp packages
+export npm_config_build_from_source=true
+
+rm $PREFIX/bin/node
+ln -s $BUILD_PREFIX/bin/node $PREFIX/bin/node
+
+NPM_CONFIG_USERCONFIG=/tmp/nonexistentrc
```
Comment (xhochy) [inline recipes/speedscope/build.sh:14]:
```suggestion
export NPM_CONFIG_USERCONFIG=/tmp/nonexistentrc
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by moritzwilksch** [comment]
where does npm install the package? Do we need to move some things around?

File: recipes/speedscope/build.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+fi
+# Don't use pre-built gyp packages
+export npm_config_build_from_source=true
+
+rm $PREFIX/bin/node
+ln -s $BUILD_PREFIX/bin/node $PREFIX/bin/node
+
+export NPM_CONFIG_USERCONFIG=/tmp/nonexistentrc
+pnpm import
+pnpm install --prod
```
Comment (xhochy) [inline recipes/speedscope/build.sh:16]: Seems like we need all the dependencies.
```suggestion
pnpm install
```

**Comment by xhochy** [comment]
> where does npm install the package? Do we need to move some things around?

It should put it into $PREFIX with the binary in `$PREFIX/bin`. Once the jobs is successful, you can look into the CI logs for the content of the package.

**Comment by moritzwilksch** [comment]
according to [this](https://github.com/protobufjs/protobuf.js/issues/1104) we might need to patch `package.json` and `scripts/build-release.sh`

**Comment by xhochy** [comment]
CI needs a compiler, but the osx job has the same error as you probably have locally. @JannikWibkerQC can probably help you do the patching correctly.

**Comment by JannikWibkerQC** [comment]
`speedscope` has a `prepack` script which is a hook being executed whenever `(p)npm pack` (or `(p)npm publish`), thus running `pnpm pack` triggers a build. I tried it locally but it fails. I'd maybe suggest just installing the package using `npm install -g speedscope@{PKG_VERSION}` as this won't run into those issues.

Afterwards you'd still do `pnpm install --prod` and everything pnpm-licenses related.


Not sure what the opinion is on installing from the npm registry vs. building it manually and installing it from there though. [`pnpm`](https://github.com/conda-forge/pnpm-feedstock/blob/main/recipe/build.sh#L24), [`prettier`](https://github.com/conda-forge/prettier-feedstock/blob/main/recipe/build.sh#L19) and [`marp-cli`](https://github.com/conda-forge/marp-cli-feedstock/blob/main/recipe/build.sh#L17
) all install from the registry while [`pnpm-licenses`](https://github.com/conda-forge/pnpm-licenses-feedstock/blob/main/recipe/build.sh#L24) gets build using conda-build.

It is to be noted that building pnpm-licenses using `build.sh` / `bld.bat` was made easier by writing the tool itself and knowing how the build system works 😅 


If you go with the npm registry approach this should work:

```bash
#!/bin/sh

set -exuo pipefail

if [[ "${target_platform}" == "osx-arm64" ]]; then
    export npm_config_arch="arm64"
fi
# Don't use pre-built gyp packages
export npm_config_build_from_source=true

rm $PREFIX/bin/node
ln -s $BUILD_PREFIX/bin/node $PREFIX/bin/node

# install speedscope from the npm registry
# all things coming after this are just concerned with generating the third-party-licenses file
export NPM_CONFIG_USERCONFIG=/tmp/nonexistentrc
npm install -g speedscope@${PKG_VERSION}

pnpm import
pnpm install --prod

# generate third party licenses file
pnpm licenses list --json --prod | pnpm-licenses generate-disclaimer --prod --json-input --output-file=third-party-licenses.txt
```

Like @xhochy mentioned you might need a c compiler:

```yaml
requirements:
  build:
    - {{ compiler('c') }}
    - {{ stdlib("c") }}
    - {{ compiler('cxx') }}
    - make
    - nodejs
```

**Comment by moritzwilksch** [comment]
@conda-forge-admin, please rerender

**Review by xhochy — APPROVED** [review]
If you can build from the source, we would prefer that as it is more self-contained (well npm packages are never self-contained).

**Pull request merged**